Chatkit/Redux/State/UserState.swift

```diff
@@ -0,0 +1,15 @@
+
+enum UserState {
+    
+    case empty
```

Comment:
I not convinced that `empty` belongs here otherwise when we implement the list of users for a room it will be possible to have `empty` users in that list, which makes no sense.

We obviously need something to model the "no current user" state.  Would it be better to drop `case empty` here and just make `ChatState.currentUser` an optional?

Reply:
Sure, that's an option. I am also not convinced about naming the two cases below as `populated` and `partial`.

Reply:
Not sure I have a strong opinion on the `populated` and `partial` bit at the moment.  I'd be happy for that to stay as it in this PR and we can address later when we are coding the actions/reducers that work with partial models.

Reply:
I replaced `empty` with an optional value.

Reply:
For reference, `.empty` is now back as per discussion here:
https://pusher.slack.com/archives/CSVSXJNEM/p1581509086001500